### DISCUSSION TITLE
Peer dependency problem for mobx-react@6.3.1

### DISCUSSION THREAD

**Discussion by Iliyan7** [mobxjs/mobx#3800] [General]
Hello,

I found something that might be an issue, but I wonder if I'm missing something. That's why I'm here to ask.

For mobx-react@6.3.1 (https://www.npmjs.com/package/mobx-react/v/6.3.1), it's clearly written that the supported React version is 16.8.0 and higher. But when I tried to install it with react@17.0.2, I got a peer dependency error, which I need clarification on.

```npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"17.0.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0 || 16.9.0-alpha.0" from mobx-react@6.3.1
npm ERR! node_modules/mobx-react
npm ERR!   mobx-react@"6.3.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/il.ivanov/.npm/eresolve-report.txt for a full report.
```
Does this seem to be expected?
And is it safe to use the `--legacy-peer-deps` flag in this case?

**Comment by micnem** [comment]
Same issue here.

I am on React 17.0.1:

![CleanShot 2024-02-12 at 16 02 03@2x](https://github.com/mobxjs/mobx/assets/29963598/58982a26-f8b2-40f8-a8a5-752c0c49ccd7)
